### PR TITLE
Sgroup

### PR DESCRIPTION
**Pull request by jones-gareth**
#### What does this implement/fix? Explain your changes.

Adds SubstanceGroup to the SWIG wrappers so we can work with V3000 mol blocks

#### Any other comments?

Also added overloads for QueryAtom::setQuery

### CODE REVIEW

File: Code/GraphMol/SubstanceGroup.h [hunk 1]
```diff
@@ -82,8 +82,9 @@ class RDKIT_GRAPHMOL_EXPORT SubstanceGroup : public RDProps {
     }
   };
 
+  // Unfortunately, SWIG generated wrapper code uses temporary variables that require a default ctor not be deleted.
   //! No default constructor
-  SubstanceGroup() = delete;
+  // SubstanceGroup() = delete;
```
Comment (bp-kelley) [inline Code/GraphMol/SubstanceGroup.h:87]: You might be able to use #ifndef SWIG here

**Review by bp-kelley — CHANGES_REQUESTED** [review]

**Comment by bp-kelley** [comment]
We should add java and c# tests here as well.

**Comment by jones-gareth** [comment]
@bp-kelley I can add a Java test easily enough, but there are currently no tests for the C# wrappers in RDKit.   Before adding any C# tests I think you'd at least need to update the C# compiler from mono to dotnet and put in place xunit testing.  That's quite a bit of CMake work and is probably best to do in another PR.  I can remove SubstanceGroup.i from the C# wrappers if not having a test is a problem.

**Comment by greglandrum** [comment]
> I can remove SubstanceGroup.i from the C# wrappers if not having a test is a problem.

My two cents: it's not a problem given that we don't have convincing C# tests for anything else

It'd be great to put C# tests in place for everything (and to improve the java tests), but that's clearly not for this PR

File: Code/GraphMol/SubstanceGroup.h (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/SubstanceGroup.h:87]: agreed that this should only be removed for SWIG builds

**Comment by bp-kelley** [comment]
There are some c# tests, I’ve been meaning to add more.  This might be a good time to start.

**Comment by jones-gareth** [comment]
I've added a Java test.  It was worth it as I caught a "free(): invalid pointer error" from SubstanceGroup#getAtoms()- the SWIG vector wrapper was freeing the returned reference.
@bp-kelley I don't really want to add any C# tests to the PR as the work is being done for Schrodinger and that's not something they need.  I put in the change to the C# wrapper as it was trivial.  I understand if you're not comfortable with the lack of testing, but, if so, I should just revert the change to the C# wrapper and do the tests in another PR.
I don't see any C# tests, just the main methods in test.cs and RDKitCSharpTest/Program.cs.  Am I missing something?

File: Code/JavaWrappers/SubstanceGroup.i [hunk 2]
```diff
@@ -1,8 +1,13 @@
+//
+// Created by Gareth Jones on 9/2/2020.
+//
+// Copyright 2020 Schrodinger, Inc
```
Comment (greglandrum) [inline Code/JavaWrappers/SubstanceGroup.i:4]: Just to make it explicit, please add the usual bit of "this is part of the RDKit" boilerplate to this too:
```suggestion
// Copyright 2020 Schrodinger, Inc
//  @@ All Rights Reserved @@
//  This file is part of the RDKit.
//  The contents are covered by the terms of the BSD license
//  which is included in the file license.txt, found at the root
//  of the RDKit source tree.
```

File: Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SubstanceGroupTests.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
+//
+// Created by Gareth Jones on 9/4/2020.
+//
+// Copyright 2020 Schrodinger, Inc
```
Comment (greglandrum) [inline Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SubstanceGroupTests.java:4]: Same comment about the boilerplate here

Comment (jones-gareth) [reply Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SubstanceGroupTests.java:4]: I've added the boilerplate- I've also updated the files I created for the tautomer search.

**Comment by bp-kelley** [comment]
@jones-gareth  I can hold off on the C# test I haven't added them for my own stuff anyway!

**Review by bp-kelley — APPROVED** [review]

File: Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SubstanceGroupTests.java (hunk 3 above)
Comment (greglandrum) [reply Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SubstanceGroupTests.java:4]: thanks. Sorry I missed those at the time

**Review by greglandrum — APPROVED** [review]

**Pull request merged**